src/Tests/ClientConcepts/HighLevel/Inferrence/IndexNameInference.doc.cs

```diff
@@ -0,0 +1,76 @@
+﻿using Elasticsearch.Net;
+using FluentAssertions;
+using Nest;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Tests.Framework;
+using Tests.Framework.MockData;
+using Xunit;
+
+namespace Tests.ClientConcepts.HighLevel.Inferrence
+{
+	public class IndexNameInference
+	{
+		[U]
+		public void DefaultIndexIsInferred()
+		{
+			var settings = new ConnectionSettings()
+				.DefaultIndex("defaultindex");
+			var resolver = new IndexNameResolver(settings);
+			var index = resolver.Resolve<Project>();
+			index.Should().Be("defaultindex");
+		}
+
+		[U]
+		public void ExplicitMappingIsInferred()
+		{
+			var settings = new ConnectionSettings()
+				.MapDefaultTypeIndices(m => m
+					.Add(typeof(Project), "projects")
+				);
+			var resolver = new IndexNameResolver(settings);
+			var index = resolver.Resolve<Project>();
+			index.Should().Be("projects");
+		}
+
+		[U]
+		public void ExplicitMappingTakesPrecedence()
+		{
+			var settings = new ConnectionSettings()
+				.DefaultIndex("defaultindex")
+				.MapDefaultTypeIndices(m => m
+					.Add(typeof(Project), "projects")
+				);
+			var resolver = new IndexNameResolver(settings);
+			var index = resolver.Resolve<Project>();
+			index.Should().Be("projects");
+		}
+
+		[U]
+		public void LeadingUpperCaseThrowsResolveException()
+		{
+			var settings = new ConnectionSettings()
+				.DefaultIndex("Default")
+				.MapDefaultTypeIndices(m => m
+					.Add(typeof(Project), "Myprojects")
+				);
+
+			var resolver = new IndexNameResolver(settings);
+
+			Assert.Throws<ResolveException>(() => resolver.Resolve<Project>());
+			Assert.Throws<ResolveException>(() => resolver.Resolve<Tag>()); // should infer default index
+			Assert.Throws<ResolveException>(() => resolver.Resolve("Foo"));
+		}
+
+		[U]
+		public void NoIndexThrowsResolveException()
+		{
+			var settings = new ConnectionSettings();
+			var resolver = new IndexNameResolver(settings);
+			Assert.Throws<ResolveException>(() => resolver.Resolve<Project>());
```

Comment:
We should assert that the message makes sense too e.g in this case no default global index and no configured index for project directly were specified on ConnectionSettings